consensus/parlia/parlia.go

```diff
@@ -874,6 +885,22 @@ func (p *Parlia) Seal(chain consensus.ChainHeaderReader, block *types.Block, res
 	return nil
 }
 
+func (p *Parlia) shouldWaitForCurrentBlockProcess(chain consensus.ChainHeaderReader, header *types.Header, snap *Snapshot) bool {
+	if snap.inturn(p.val) {
+		return false
+	}
+
+	highestVerifiedHeader := chain.GetHighestVerifiedHeader()
+	if highestVerifiedHeader == nil {
+		return false
+	}
+
+	if header.ParentHash == highestVerifiedHeader.ParentHash && header.Difficulty.Cmp(highestVerifiedHeader.Difficulty) <= 0 {
```

Comment:
`header.Difficulty.Cmp(highestVerifiedHeader.Difficulty) <= 0` is not needed, because if `header.Difficulty.Cmp(highestVerifiedHeader.Difficulty) > 0` means it is inturn block, it should return in previous logic.